src/components/application_manager/src/rpc_passing_handler.cc

```diff
@@ -93,6 +93,63 @@ bool RPCPassingHandler::CanHandleFunctionID(int32_t function_id) {
   return false;
 }
 
+bool RPCPassingHandler::IsPassthroughAllowed(
+    smart_objects::SmartObject rpc_message) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  mobile_api::FunctionID::eType function_id =
+      static_cast<mobile_api::FunctionID::eType>(
+          rpc_message[strings::params][strings::function_id].asInt());
+  uint32_t connection_key =
+      rpc_message[strings::params][strings::connection_key].asUInt();
+  std::string function_id_str =
```

Comment:
@jacobkeeler 
In case if we need function_id only to get string function id, I would not split this expressions by retrieving connection_key.